src/whois.cc

```diff
@@ -172,6 +172,8 @@ WhoisState::readReply(const Comm::ConnectionPointer &conn, char *aBuffer, size_t
 
     if (dataWritten) // treat zero-length responses as incomplete
         fwd->markStoredReplyAsWhole("whois received/stored the entire response");
+    else
+        fwd->fail(new ErrorState(ERR_ZERO_SIZE_OBJECT, Http::scBadGateway, fwd->request, fwd->al));
```

Comment:
If possible, please address the XXX in the PR description devoted to this part of the fix. Were you able to test this change?

Please also review my PR title/description changes and adjust further if needed. My edits were based on our truncated-responses.js test.

Reply:
> address the XXX in the PR description

Addressed.

> Were you able to test this change?

I only checked that cache.log now has several entries with ERR_ZERO_SIZE_OBJECT. WHOIS is used only during startup, by src_as/dst_as ACLs, so access.log is out of scope.

> review my PR title/description changes 

Done.

Reply:
Thank you. If WHOIS is also used during reconfiguration, please change "startup" in the PR description to "(re)configuration".